pkg/skaffold/docker/image.go

```diff
@@ -160,6 +160,10 @@ func (l *localDaemon) ConfigFile(ctx context.Context, image string) (*v1.ConfigF
 func (l *localDaemon) Build(ctx context.Context, out io.Writer, workspace string, a *latest.DockerArtifact, ref string, mode config.RunMode) (string, error) {
 	logrus.Debugf("Running docker build: context: %s, dockerfile: %s", workspace, a.DockerfilePath)
 
+	if a.Secret != nil {
+		return "", fmt.Errorf("docker build secrets require BuildKit - set `useBuildkit: true` in your config, or run with `DOCKER_BUILDKIT=1`")
+	}
```

Comment:
It seems slightly odd to have this string directly in `Build()`.  It might be better to push this into a `localDaemon.CheckCompatible(latest.DockerArtifact)`.

Or alternatively, maybe the docker builder should automatically switch to BuildKit mode?  BuildKit is now the default in D4D edge:
https://github.com/GoogleContainerTools/skaffold/blob/7b4eb4a9fe51edc372e5f0ef15cdeb71a83a9fb6/pkg/skaffold/build/local/docker.go#L49-L53

Reply:
yeah fair enough, I'll move this out into a helper. I'll keep any changes that switch build modes out of this PR, but would be interesting to investigate switching to buildkit automatically in docker builds in skaffold.